pulsar-client/src/main/java/org/apache/pulsar/client/impl/PartitionedProducerImpl.java

```diff
@@ -244,4 +268,102 @@ String getHandlerName() {
         return "partition-producer";
     }
 
+    // This listener is triggered when topics partitions are updated.
+    private class TopicsPartitionChangedListener implements PartitionsChangedListener {
+        // Check partitions changes of passed in topics, and add new topic partitions.
+        @Override
+        public CompletableFuture<Void> onTopicsExtended(Collection<String> topicsExtended) {
+            CompletableFuture<Void> future = new CompletableFuture<>();
+            if (topicsExtended.isEmpty() || !topicsExtended.contains(topic)) {
+                future.complete(null);
+                return future;
+            }
+
+            client.getLookup().getPartitionedTopicMetadata(TopicName.get(topic)).thenCompose(metadata -> {
```

Comment:
`PulsarClient` has already a method that returns the list of partitions for the topic in standard way. Maybe that could be reused here as well, instead of constructing the names with `String partitionName = TopicName.get(topic).getPartition(partitionIndex).toString();`